File: ios/Classes/Features/AgentStart.swift
import ADEUMInstrumentation
import Flutter

extension SwiftAppDynamicsAgentPlugin {
  public func start(result: @escaping FlutterResult, arguments: Any?) {
    let properties = arguments as! Dictionary<String, Any>

    guard let appKey = properties["appKey"] as? String else {
      let error = FlutterError(code: "500", message: "Agent start() failed.", details: "Please provide an appKey.")
      result(error)
      return
    }
    
    let configuration = ADEumAgentConfiguration(appKey: appKey)
    let type = properties["type"] as! String
    let version = properties["version"] as! String
    
    if let loggingLevel = properties["loggingLevel"] as? Int {
      let levels = [ADEumLoggingLevel.off,  ADEumLoggingLevel.info, ADEumLoggingLevel.all]
      configuration.loggingLevel = levels[loggingLevel]
    }
    
    if let anrDetectionEnabled = properties["anrDetectionEnabled"] as? Bool {
      configuration.anrDetectionEnabled = anrDetectionEnabled
    }
    
    if let anrStackTraceEnabled = properties["anrStackTraceEnabled"] as? Bool {
      configuration.anrStackTraceEnabled = anrStackTraceEnabled
    }
    
    if let collectorURL = properties["collectorURL"] as? String {
      configuration.collectorURL = collectorURL
    }
    
    if let screenshotURL = properties["screenshotURL"] as? String {
      configuration.screenshotURL = screenshotURL
    }
    
    if let screenshotsEnabled = properties["screenshotsEnabled"] as? Bool {
      configuration.screenshotsEnabled = screenshotsEnabled
    }
    
    if let crashReportingEnabled = properties["crashReportingEnabled"] as? Bool {
      configuration.crashReportingEnabled = crashReportingEnabled
    }
    
    if let applicationName = properties["applicationName"] as? String {
      configuration.applicationName = applicationName
    }
    
    if crashReportCallback == nil {
      crashReportCallback = CrashCallbackObject()
      configuration.crashReportCallback = crashReportCallback
    }
    
    if let enableLoggingInVSCode = properties["enableLoggingInVSCode"] as? Bool {
      configuration.enableLoggingInVSCode = enableLoggingInVSCode
    }
    
    configuration.enableAutoInstrument = false
    configuration.jsAgentEnabled = false
    configuration.jsAgentAjaxEnabled = false
    configuration.jsAgentFetchEnabled = false
    configuration.jsAgentZonePromiseEnabled = false

    ADEumInstrumentation.initWith(configuration, a: type, b: version);
    result(nil)
    }
}

File: android/src/main/kotlin/com/appdynamics/appdynamics_agent/features/AgentStart.kt
/*
 * Copyright (c) 2022. AppDynamics LLC and its affiliates.
 * All rights reserved.
 *
 */

package com.appdynamics.appdynamics_agent.features

import androidx.annotation.NonNull
import com.appdynamics.appdynamics_agent.AppDynamicsAgentPlugin
import com.appdynamics.eumagent.runtime.AgentConfiguration
import com.appdynamics.eumagent.runtime.Instrumentation
import io.flutter.plugin.common.MethodChannel

fun AppDynamicsAgentPlugin.start(@NonNull result: MethodChannel.Result, arguments: Any?) {
    val properties = arguments as HashMap<*, *>
    val agentVersion = properties["version"] as String
    val agentName = properties["type"] as String
    val appKey = properties["appKey"] as? String
    val loggingLevel = properties["loggingLevel"] as? Int
    val collectorURL = properties["collectorURL"] as? String
    val screenshotURL = properties["screenshotURL"] as? String
    val screenshotsEnabled = properties["screenshotsEnabled"] as? Boolean
    val crashReportingEnabled = properties["crashReportingEnabled"] as? Boolean
    val applicationName = properties["applicationName"] as? String

    if (appKey == null) {
        result.error("500", "Please provide an appKey.", "Agent start() failed.")
        return
    }

    val builder: AgentConfiguration.Builder =
        AgentConfiguration.builder()
            .withAppKey(appKey)

    if (loggingLevel != null) {
        val levels = listOf(
            Instrumentation.LOGGING_LEVEL_NONE,
            Instrumentation.LOGGING_LEVEL_INFO,
            Instrumentation.LOGGING_LEVEL_VERBOSE,
        )
        builder.withLoggingLevel(levels[loggingLevel])
    }

    if (collectorURL != null) {
        builder.withCollectorURL(collectorURL)
    }

    if (screenshotURL != null) {
        builder.withScreenshotURL(screenshotURL)
    }

    if (screenshotsEnabled != null) {
        builder.withScreenshotsEnabled(screenshotsEnabled)
    }

    if (crashReportingEnabled != null) {
        builder.withCrashReportingEnabled(crashReportingEnabled)
    }

    if (applicationName != null) {
        builder.withApplicationName(applicationName)
    }

    if (crashReportCallback == null) {
        crashReportCallback = CrashCallbackObject(channel)
        builder.withCrashCallback(crashReportCallback)
    }

    builder
        .withAutoInstrument(false)
        .withContext(context)
        .withJSAgentInjectionEnabled(false)
        .withJSAgentAjaxEnabled(false)
    Instrumentation.startFromHybrid(builder.build(), agentName, agentVersion)

    result.success(null)
}
